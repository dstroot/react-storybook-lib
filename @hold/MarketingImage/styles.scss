/* The image box */
.img-box {
  position: relative;
  height: 425px;
}

/* Place text on the image */
.img-box-text {
  max-width: 650px;
  position: absolute;
  text-align: center;
  margin: 10%;
  margin-top: 30%;
}
@media (min-width: 768px) {
  .img-box-text {
    margin-left: 50%;
    margin-top: 15%;
    max-width: 550px;
    position: absolute;
    text-align: left;
  }
}

.img-box-content {
  font-size: calc(10px + 0.5vw + 0.5vh + 0.25vmin);
}

// darken on rollover
.img-box:hover .overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  background-color: rgba(0, 142, 215, 0.6);
  transition: 0.5s ease;
}

// images and info
// ----------------
// https://bing.biturl.top/?resolution=1920&format=image&index=0&mkt=en-US
// http://unsplash.it/1200x800
// ----------------
// https://www.google.com/intl/en/about/
// https://www.w3schools.com/css/css3_backgrounds.asp
