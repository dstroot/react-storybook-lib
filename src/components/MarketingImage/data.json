[
  {
    "image": "/media/1.jpg",
    "lead": "Plan now, so you’re ready later",
    "text": "For you, family is one of the most important things in your life. You take care of them. They take care of you. Making sure that they’ll always be taken care of, no matter what happens.",
    "url": "/"
  },

  {
    "image": "/media/2.jpg",
    "lead": "Save enough for my future",
    "text": "You don’t know what the future has planned for you, but you want to be prepared for the unexpected and be able to achieve your goals.",
    "url": "/"
  },
  {
    "image": "/media/3.jpg",
    "lead": "Protect and grow my business",
    "text": "You want your business to maintain its competitive edge. Attracting talent and building a succession plan for the future means you can ensure your business stays in stable hands.",
    "url": "/"
  }
]
