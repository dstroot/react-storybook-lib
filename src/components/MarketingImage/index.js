import React from 'react';
import PropTypes from 'prop-types';

// Styles
import "./styles.scss";

const MarketingImage = ({ lead, text, image, url }) => {
  return (
    <a href={url}>
      <div
        className="img-box mt-3"
        style={{
          background: `url(${image}) left top no-repeat`,
          backgroundSize: `cover`,
        }}
      >
        <div className="overlay">
          <div className="img-box-text text-white">
            <h3 className="text-uppercase">{lead}</h3>
            <p className="img-box-content">{text}</p>
          </div>
        </div>
      </div>
    </a>
  );
};

export default MarketingImage;

MarketingImage.propTypes = {
  image: PropTypes.string.isRequired,
  lead: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
