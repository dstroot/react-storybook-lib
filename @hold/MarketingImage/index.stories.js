import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import MarketingImage from '../MarketingImage';
import data from '../MarketingImage/data.json';
import image from './media/1.jpg';

// MarketingImage section
storiesOf('MarketingImage', module)
  .add('show MarketingImage', () => (
    <MarketingImage
      lead="Our Commitments"
      text="We are dedicated to creating opportunity for everyone"
      image={image}
      url="/"
    />
  ))
  .add('show MarketingImage list', () =>
    data.map((record, index) => (
      <MarketingImage
        key={index}
        image={record.image}
        lead={record.lead}
        text={record.text}
        url={record.url}
      />
    ))
  );
