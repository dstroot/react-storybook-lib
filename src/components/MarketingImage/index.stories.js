import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs';

// components
import MarketingImage from '../MarketingImage';
// import data from '../MarketingImage/data.json';
import image from './media/1.jpg';

const stories = storiesOf('Pac Life|MarketingImage', module);
stories.addDecorator(withKnobs);

// MarketingImage section
stories
  .addParameters({
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show MarketingImage', () => (
    <MarketingImage
      lead={text('lead', 'Our Commitments')}
      text={text(
        'subtext',
        'We are dedicated to creating opportunity for everyone'
      )}
      image={image}
      url="/"
    />
  ));
// .add('show MarketingImage list', () =>
//   data.map((record, index) => (
//     <MarketingImage
//       key={index}
//       image={record.image}
//       lead={record.lead}
//       text={record.text}
//       url={record.url}
//     />
//   ))
// );
