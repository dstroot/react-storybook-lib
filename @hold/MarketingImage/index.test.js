import React from 'react';
import renderer from 'react-test-renderer';
import MarketingImage from '../MarketingImage';

describe('MarketingBing', () => {
  it('it should render', () => {
    const component = renderer.create(
      <MarketingImage
        lead="Our Commitments"
        text="We are dedicated to creating opportunity for everyone"
        image="/media/1.jpg"
        url="/"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
