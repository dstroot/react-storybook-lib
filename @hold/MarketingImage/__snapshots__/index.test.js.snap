// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingBing it should render 1`] = `
<a
  href="/"
>
  <div
    className="img-box mt-3"
    style={
      Object {
        "background": "url(/media/1.jpg) left top no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="overlay"
    >
      <div
        className="img-box-text text-white"
      >
        <h3
          className="text-uppercase"
        >
          Our Commitments
        </h3>
        <p
          className="img-box-content"
        >
          We are dedicated to creating opportunity for everyone
        </p>
      </div>
    </div>
  </div>
</a>
`;
